# Coding Exercises (Part 1)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code! 

The correct answer is provided below your coding cell. There you can check whether your code is correct.

If you need some help or if you want to check your code, you can also have a look at the __solutions__.

### Have Fun!

--------------------------------------------------------------------------------------------------------------

## Exercise 1: Using Python as a Calculator for TVM Problems

Use __variables__ where appropriate! __Input__ values and your __final results__ shall be stored in variables. 

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [1]:
# Store Input values in variables
PV = 2000 # present value
i = 0.04 #interest rate
n = 7 #period


In [2]:
# Calculate and store the future value in a variable
FV = PV*(1+i)**n

In [3]:
# print the future value
FV

2631.8635584716803

(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [18]:
# Store Input values in variables
FV = 20000 #future value
n = 4 #period
i = 0.035 #interest rate

In [19]:
# Calculate and store the present value in a variable
PV = FV/(1+i)**n

In [20]:
# print the present value
PV

17428.84455397145

(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [51]:
#Bank B
PV = 10000
FV = 15000
n = 10

i = (FV / PV)**(1/n) - 1

In [52]:
i = i * 100
i

4.137974399241062

In [ ]:
#I will accept the offer from Bank A because Bank A offer a higher rate that Bank B

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [66]:
P_old = 976
Div = 40
P_new = 1053

In [71]:
P_r = (P_new / P_old - 1) * 100

In [72]:
P_r

7.889344262295084

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [75]:
Div_r = (Div/ P_old) * 100

In [76]:
Div_r

4.098360655737705

In [77]:
Total_Return = P_r + Div_r

In [78]:
Total_Return

11.987704918032788

(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [81]:
print(P_r, Div_r, Total_Return, sep = ', ')

7.889344262295084, 4.098360655737705, 11.987704918032788


# Well Done!

---------------------------------------------------------------------------------------------------------------------

# Solutions (Stop here if you want to code on your own!)

# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [ ]:
# Store Input values in variables
pv = 2000
n = 7
r = 0.04

In [ ]:
# Calculate and store the future value in a variable
fv = pv * (1 + r)**n

In [ ]:
# print the future value
fv

(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [ ]:
# Store Input values in variables
fv = 20000
n = 4
r = 0.035

In [ ]:
# Calculate and store the present value in a variable
pv = fv / (1 + r)**n

In [ ]:
# print the present value
pv

(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [ ]:
# Store Input values in variables
pv = 10000
fv = 15000
n = 10

In [ ]:
r = (fv / pv)**(1/n) - 1

In [ ]:
r

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [ ]:
p0 = 976
p1 = 1053
div = 40

In [ ]:
p_ret = p1 / p0 - 1

In [ ]:
p_ret

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [ ]:
div_y = div / p0

In [ ]:
div_y

In [ ]:
total_ret = p_ret + div_y

In [ ]:
total_ret

(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [ ]:
print(p_ret, div_y, total_ret, sep = ", ")